src/types.ts

```diff
@@ -214,7 +270,11 @@ export interface PostHogConfig {
     feature_flag_request_timeout_ms: number
     get_device_id: (uuid: string) => string
     name: string
+    /**
+     * this is a read-only function that can be used to react to event capture
+     */
     _onCapture: (eventName: string, eventData: CaptureResult) => void
+    beforeCapture: (cr: CaptureResult) => CaptureResult | null
```

Comment:
Semantically speaking, the fn is executed after capture (the fn) or the act of capturing an event.
And the fn is executed before enqueuing/flushing the event.
I wonder if the naming would not cause any confusion.
`beforeEnqueuing` is strange but sounds semantically correct, anyway, just giving food for thought, your call =)

Reply:
`beforeSend`??

Reply:
that makes sense to me, sentry uses [beforeSend](https://docs.sentry.io/platforms/javascript/configuration/filtering/#using-before-send) as well